python/rapidsmpf/rapidsmpf/streaming/core/context.pyx

```diff
@@ -71,3 +76,35 @@ cdef class Context:
 
     def statistics(self):
         return self._statistics
+
+    def get_stream_from_pool(self) -> Stream:
+        """
+        Get a stream from the stream pool.
+
+        Returns
+        -------
+        Stream
+            A stream from the stream pool.
+        """
+        cdef const cuda_stream_pool* pool_ptr = self._br.stream_pool()
+        cdef cuda_stream_view stream_view
+        with nogil:
+            stream_view = pool_ptr.get_stream()
```

Comment:
*nit*: It's not worth dropping the gil here, this function is just an atomic increment and an array index.